### ISSUE TITLE
Request for Examples of next.js App Router (App Directory)

### ISSUE THREAD

**Issue by Yedidya10** [nextauthjs/next-auth#7953]
### What is the improvement or update you wish to see?

Hi,

A lot of things have changed and improved in the new version of next js (13.2) but the documents that can be found for next-auth are few and I found myself getting confused with it.  I think it would be great to create examples of this library in the app router folder.

### Is there any context that might help us understand?

If you go to the apps>examples>nextjs folder you can see that there are examples only for the page router version and not for app router.

**Comment by zwhitchcox** [comment]
It looks like those repo's are using an API that's not in production yet..when I try to use it:

```
- error app/page.tsx (7:28) @ auth
- error TypeError: (0 , _auth__WEBPACK_IMPORTED_MODULE_1__.auth) is not
 a function
    at Home (./app/page.tsx:19:70)
    at async Promise.all (index 0)
   5 | 
   6 | export default async function Home() {
>  7 |   const session = await auth();
     |                            ^
   8 |   return (
   9 |     <>
  10 |       <Head>
```

and

```
  31 | 
  32 | export const {
> 33 |   handlers: { GET, POST },
     |              ^
  34 |   auth,
  35 |   CSRF_experimental,
  36 | } = NextAuth(authOptions)

```

Do you have any suggestions on how to handle this, @balazsorban44 ? Like should we be using a specific release? Or is it better to stick to the pages router for now and just wait for it to be more stable? I'm starting a new project, so I didn't want to have to rebuild everything with the App Router later, so I figured I'd save some time by just implementing it now, but it looks like it might not be ready.

**Comment by pschaub** [comment]
According to the current documentation, you don't need next-auth v5 to use next-auth with Next.js 13 app router.

Have you seen this part in the v4 documentation?
> If you're using [Next.js 13.2](https://nextjs.org/blog/next-13-2#custom-route-handlers) or above with the new App Router (app/), you can initialize the configuration using the new [Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/router-handlers) by following our [guide](https://next-auth.js.org/configuration/initialization#route-handlers-app).

Source: https://next-auth.js.org/getting-started/example#add-api-route

There is an example how to use next-auth v4 with Next.js 13.2 app router (route handlers) here: https://next-auth.js.org/configuration/initialization#route-handlers-app

Have you tried this?

**Comment by Yedidya10** [comment]
Yes, I saw it, but the code examples are partial and it's hard for me to understand how I'm supposed to implement signIn with getProviders in the new fetch method. 

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching

Some of the variations I tried didn't work, and some times the component would render a lot, like 15-20 times.  (I prefer to use a server component rather than a client component)

**Comment by chrisivo** [comment]
It looks like `NextAuth(authOptions)` now returns an async function, rather than an object containing `auth` and `handler`. Is there a v5 release I can roll back to which returns the object as specified currently in the documentation?

**Comment by vegandiet705** [comment]
@pschaub I'm trying to launch the example app from [here](https://github.com/nextauthjs/next-auth/tree/790f7617083d055b94d05c5c815749258da84852/apps/dev/nextjs). I followed the instruction from [here](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md#setting-up-local-environment). When I try to launch the app, it gives 404 error. In VS code, it gives the error:

`Cannot find module 'next-auth/next' or its corresponding type declarations.`

Though, my local node_modules seems to have this package. I don't understand. It's coming from the import in `app/server-component/page.tsx`. Any idea on how to fix it ?

**Comment by vegandiet705** [comment]
I'm also interested in more examples on how to use NextAuth.js with Next.js App Router. I just started using Next.js. I passed some days reading all the docs about the App Router and built my entire app using it. Now it's only remaining the authentication. It will mess with my schedule to have to migrate to the old Next.js to be able to use NextAuth.js :(
